### PR TITLE
cli: stop alarm when cli is open

### PR DESCRIPTION
**Pull request by yaronm-hailo**
The debugger cli uses an alarm that periodically read a character from UART
to check if ctrl+E was pressed. However, the alarm is still being called
when the CLI is opened - which causes the CLI to occasionally lose characters.

### CODE REVIEW

**Comment by tarek-arm** [comment]
Jenkins: OK to test

File: module/debugger_cli/src/mod_debugger_cli.c [hunk 1]
```diff
@@ -103,7 +119,9 @@ static int debugger_cli_process_event(const struct fwk_event *event,
     case DEBUGGER_CLI_INTERNAL_EVENT_IDX_ENTER_DEBUGGER:
         /* Start the CLI, blocking the rest of the event queue */
         cli_print("[CLI_DEBUGGER_MODULE] Entering CLI\n");
+        stop_alarm(event->target_id);
```
Comment (nicola-mazzucato-arm) [inline module/debugger_cli/src/mod_debugger_cli.c:122]: Please make sure you use the returned value from start_ & stop_ functions.
Thanks for the improvements :)

**Review by nicola-mazzucato-arm — CHANGES_REQUESTED** [review]

File: module/debugger_cli/src/mod_debugger_cli.c (hunk 1 above)
Comment (yaronm-hailo) [reply module/debugger_cli/src/mod_debugger_cli.c]: I pushed a new version with checks for the return values of the start & stop functions

**Comment by tarek-arm** [comment]
Jenkins: OK to test

**Review by tarek-arm — APPROVED** [review]

**Pull request merged**